sunpy/map/mapbase.py

```diff
@@ -286,7 +286,7 @@ def _repr_html_(self):
             bad_pixel_text += ", ".join(text_list)
 
         # Use a grayscale colormap with histogram equalization (and red for bad values)
-        cmap = cm.get_cmap('gray')
+        cmap = copy.copy(cm.get_cmap('gray'))
```

Comment:
We copy to prevent modification?

Reply:
This is because we do `.set_bad` on the line below, which without copying would modify the default copy of the colormap in matplotlib, which raises a warning now. I'll add a comment to the code saying why we copy.